components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java

```diff
@@ -183,6 +183,9 @@ public RefreshTokenValidationDataDO validateRefreshToken(String consumerKey, Str
                     String authenticatedIDP = null;
                     if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
                         authenticatedIDP = resultSet.getString(12);
+                        validationDataDO.setTokenBindingReference(resultSet.getString(13));
```

Comment:
If we add TOKEN_BINDING_REF in the select statement before IDP.NAME in query RETRIEVE_ACCESS_TOKEN_VALIDATION_DATA_IDP_NAME_* queries, we can skip this check for TOKEN_BINDING_REF column.

Reply:
fixed.